diff --git a/CMakeLists.txt b/CMakeLists.txt
index d20a5ea..18c77c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
-cmake_minimum_required (VERSION 3.1)
-project (RingDecomposerLib VERSION 1.0)
+cmake_minimum_required(VERSION 3.1)
+project(RingDecomposerLib VERSION 1.1.3)
 
 enable_testing()
 
@@ -34,9 +34,6 @@ set(SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
 set(LIB_SOURCE_DIR "${SOURCE_DIR}/RingDecomposerLib")
 set(PYTHON_SOURCE_DIR "${SOURCE_DIR}/python")
 
-set(SHARED_LIBRARY "RingDecomposerLib")
-set(STATIC_LIBRARY "RingDecomposerLibStatic")
-
 set(SHARED_TEST "Test")
 set(STATIC_TEST "Test_static")
 
@@ -65,3 +62,36 @@ if(BUILD_PYTHON_WRAPPER)
 endif()
 
 add_subdirectory("${TEST_DIR}")
+
+target_include_directories(RingDecomposerLib PUBLIC $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/RingDecomposerLibConfigVersion.cmake"
+  VERSION 1.1.3
+  COMPATIBILITY AnyNewerVersion
+)
+
+configure_package_config_file(
+  "config.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/RingDecomposerLibConfig.cmake"
+  INSTALL_DESTINATION "lib/cmake/RingDecomposerLib"
+)
+
+install(
+  FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/RingDecomposerLibConfigVersion.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/RingDecomposerLibConfig.cmake"
+  DESTINATION "lib/cmake/RingDecomposerLib"
+)
+
+export(
+  EXPORT RingDecomposerLibTargets
+  FILE "${CMAKE_CURRENT_BINARY_DIR}/RingDecomposerLibTargets.cmake"
+)
+
+install(
+  EXPORT RingDecomposerLibTargets
+  FILE "RingDecomposerLibTargets.cmake"
+  DESTINATION "lib/cmake/RingDecomposerLib"
+)
diff --git a/config.cmake.in b/config.cmake.in
new file mode 100644
index 0000000..0323277
--- /dev/null
+++ b/config.cmake.in
@@ -0,0 +1,3 @@
+include(${CMAKE_CURRENT_LIST_DIR}/RingDecomposerLibTargets.cmake)
+
+@PACKAGE_INIT@
diff --git a/src/RingDecomposerLib/CMakeLists.txt b/src/RingDecomposerLib/CMakeLists.txt
index f032c12..579bcf3 100644
--- a/src/RingDecomposerLib/CMakeLists.txt
+++ b/src/RingDecomposerLib/CMakeLists.txt
@@ -1,31 +1,28 @@
-function(build_library LIBRARY_NAME SPECIFIER)
-  add_library("${LIBRARY_NAME}" "${SPECIFIER}"
-              RDLutility.c
-              RDLhandler.c
-              RDLrelation.c
-              RDLapsp.c
-              RingDecomposerLib.c
-              RDLgraph.c
-              RDLinfo.c
-              RDLcycleFams.c
-              RDLstack.c
-              RDLtesting.c
-              RDLtarjan.c
-              RDLdimacs.c
-              RDLbitset.c)
-  set_property(TARGET "${LIBRARY_NAME}" PROPERTY C_STANDARD 90)
-  set_target_properties("${LIBRARY_NAME}" PROPERTIES POSITION_INDEPENDENT_CODE True)
-  if(NOT WIN32)
-    target_link_libraries("${LIBRARY_NAME}" m)
-  endif()
-  install(TARGETS ${LIBRARY_NAME}
-          ARCHIVE DESTINATION lib
-          LIBRARY DESTINATION lib
-          RUNTIME DESTINATION bin)
-endfunction()
+add_library(RingDecomposerLib
+  RDLutility.c
+  RDLhandler.c
+  RDLrelation.c
+  RDLapsp.c
+  RingDecomposerLib.c
+  RDLgraph.c
+  RDLinfo.c
+  RDLcycleFams.c
+  RDLstack.c
+  RDLtesting.c
+  RDLtarjan.c
+  RDLdimacs.c
+  RDLbitset.c)
+set_property(TARGET RingDecomposerLib PROPERTY C_STANDARD 90)
+set_target_properties(RingDecomposerLib PROPERTIES POSITION_INDEPENDENT_CODE ON)
+if(NOT WIN32)
+  target_link_libraries(RingDecomposerLib m)
+endif()
+install(
+  TARGETS RingDecomposerLib
+  EXPORT RingDecomposerLibTargets
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+)
 
-build_library("${SHARED_LIBRARY}" "SHARED")
-build_library("${STATIC_LIBRARY}" "STATIC")
-
-install(FILES RingDecomposerLib.h
-        DESTINATION include)
+install(FILES RingDecomposerLib.h DESTINATION include)
diff --git a/src/Test/CMakeLists.txt b/src/Test/CMakeLists.txt
index 0397b93..7c9abb9 100644
--- a/src/Test/CMakeLists.txt
+++ b/src/Test/CMakeLists.txt
@@ -1,10 +1,3 @@
-function(build_test TEST_EXECUTABLE LIBRARY)
-  add_executable("${TEST_EXECUTABLE}" Test.c
-                 TestDemo.c TestValidate.c)
-  set (EXTRA_LIBS "${LIBRARY}")
-  set_property(TARGET "${TEST_EXECUTABLE}" PROPERTY C_STANDARD 90)
-  target_link_libraries ("${TEST_EXECUTABLE}" ${EXTRA_LIBS})
-endfunction()
-
-build_test("${SHARED_TEST}" "${SHARED_LIBRARY}")
-build_test("${STATIC_TEST}" "${STATIC_LIBRARY}")
+add_executable(Test Test.c TestDemo.c TestValidate.c)
+set_property(TARGET Test PROPERTY C_STANDARD 90)
+target_link_libraries(Test RingDecomposerLib)
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index ad27c9c..6db318a 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -2,7 +2,7 @@ file(GLOB files "${TEST_DIR}/*.dimacs")
 foreach(filename ${files})
     get_filename_component(tname "${filename}" NAME_WE)
     add_test(NAME "test_${tname}" WORKING_DIRECTORY "${TEST_DIR}"
-             COMMAND "${STATIC_TEST}" "validate" "${filename}" "200")
+             COMMAND Test "validate" "${filename}" "200")
 endforeach()
 
 if(BUILD_MINIMAL_EXAMPLE)
